### PR TITLE
docs: update development basics regarding 'make update'

### PR DESCRIPTION
**Pull request by SvenRoederer**
based on #2212  I added some sentences why ```make update``` has to be called explicitly and not by an dependency of ```make all```.

### CODE REVIEW

File: docs/dev/basics.rst [hunk 1]
```diff
@@ -37,6 +37,9 @@ rerun
 `make update` also applies the patches that can be found in the directories found in
 `patches`; the resulting branch will be called `patched`, while the commit specified in `modules`
 can be referred to by the branch `base`.
+To separately call `make update` was choosen, to let you decide when to update and reset the external
+repos. This will prevent potential loss of commits during local development amd allow building Gluon
+without the need of having an internet connection.
```
Comment (mweinelt) [inline docs/dev/basics.rst:42]: Not really, given that `make download` is a thing.

Comment (SvenRoederer) [reply docs/dev/basics.rst]: I assume you are referencing to OpenWrts make download. But that's out of Gluons Makefile ...

Anyway I updated the sentence, as I found a typo.

File: docs/dev/basics.rst [hunk 2]
```diff
@@ -37,6 +37,10 @@ rerun
 `make update` also applies the patches that can be found in the directories found in
 `patches`; the resulting branch will be called `patched`, while the commit specified in `modules`
 can be referred to by the branch `base`.
+To separately call `make update` was choosen, to let you decide when to update and reset the external
```
Comment (lemoer) [inline docs/dev/basics.rst:40]:
```suggestion
The separately call `make update` was choosen to let you decide when to update and reset the external
```

File: docs/dev/basics.rst [hunk 3]
```diff
@@ -37,6 +37,10 @@ rerun
...
+repos. This will prevent potential loss of commits during local development and allow building Gluon
```
Comment (lemoer) [inline docs/dev/basics.rst:41]:
```suggestion
repos. This avoids potential loss of commits during local development and allows building Gluon
```

File: docs/dev/basics.rst (hunk 2 above)
Comment (mweinelt) [reply docs/dev/basics.rst:40]: This sounds like you misread the sentence?

**Comment by NeoRaider** [comment]
The explanation is not very accurate, and I don't think "working with repositories" is the right section to add this information. I've opened #2356 instead.

**Pull request closed without merging**